src/Engine/Animation/AnimationController.hpp

```diff
@@ -0,0 +1,123 @@
+    /// An animation loaded from a file.
+    class AnimationController {
+        public:
+            /// Animaiton action node.
+            struct AnimationAction : public Node {
+                char animationClipName[128];
+                float playbackModifier = 1.0f;
+                bool repeat = true;
+                Animation::AnimationClip * animationClip = nullptr;
+
+                virtual ~AnimationAction() {
+                    if (animationClip != nullptr)
+                        Managers().resourceManager->FreeAnimationClip(animationClip);
+                }
+
+                /// Save the animation action node.
+                /**
+                 * @param file File to save to.
+                 */
+                virtual void Save(std::ofstream* file) override {
+                    Node::Save(file);
+                    file->write(reinterpret_cast<char*>(animationClipName), 128);
+                    file->write(reinterpret_cast<char*>(&playbackModifier), sizeof(uint32_t));
+                    file->write(reinterpret_cast<char*>(&repeat), sizeof(bool));
+                }
+
+                /// Load the animation action node.
+                /**
+                 * @param file File to load from.
+                 */
+                virtual void Load(std::ifstream* file) override {
+                    Node::Load(file);
+                    file->read(reinterpret_cast<char*>(animationClipName), 128);
+                    file->read(reinterpret_cast<char*>(&playbackModifier), sizeof(uint32_t));
+                    file->read(reinterpret_cast<char*>(&repeat), sizeof(bool));
+
+                    if (animationClip != nullptr)
+                        Managers().resourceManager->FreeAnimationClip(animationClip);
+
+                    animationClip = Managers().resourceManager->CreateAnimationClip(animationClipName);
+                }
+            };
+
+            /// Animation transition node.
+            struct AnimationTransition : public Node {
+                float transitionTime;
+
+                /// Save the animation transition node.
+                /**
+                 * @param file File to save to.
+                 */
+                virtual void Save(std::ofstream* file) override {
+                    Node::Save(file);
+                    file->write(reinterpret_cast<char*>(&transitionTime), sizeof(float));
+                }
+
+                /// Load the animation transition node.
+                /**
+                 * @param file File to load from.
+                 */
+                virtual void Load(std::ifstream* file) override {
+                    Node::Load(file);
+                    file->read(reinterpret_cast<char*>(&transitionTime), sizeof(float));
+                }
+            };
+
+            AnimationController() = default;
+            ~AnimationController();
+
+            /// Save animation controller.
+            ENGINE_API void Save(const std::string& path);
```

Comment:
Missing parameter documentation.

Reply:
This should not be included in this pull request and will be removed.